hudi/src/main/scala/io/qbeast/spark/hudi/HudiMetadataManager.scala

```diff
@@ -0,0 +1,166 @@
+    val extraOptions =
+      if (!options.extraOptions.contains(HoodieWriteConfig.TBL_NAME.key))
+        options.extraOptions ++ Map(
+          HoodieWriteConfig.TBL_NAME.key -> Paths.get(tableID.id).getFileName.toString)
+      else options.extraOptions
+    val newOptions = options.copy(extraOptions = extraOptions)
+
+    if (!existsLog(tableID)) createTable(tableID, newOptions.extraOptions)
+
+    val metaClient = loadMetaClient(tableID)
+    val metadataWriter =
+      HudiMetadataWriter(tableID, writeMode, metaClient, newOptions, schema)
+
+    metadataWriter.writeWithTransaction(writer)
+  }
+
+  override def updateMetadataWithTransaction(
+      tableID: QTableID,
+      dataSchema: StructType,
+      overwrite: Boolean)(config: => Configuration): Unit = {
+
+    val metaClient = loadMetaClient(tableID)
+    val metadataWriter =
+      HudiMetadataWriter(
+        tableID,
+        SaveMode.Append.toString,
+        metaClient,
+        QbeastOptions.empty,
+        dataSchema)
+
+    metadataWriter.updateMetadataWithTransaction(config, overwrite)
+
+  }
+
+  override def loadSnapshot(tableID: QTableID): HudiQbeastSnapshot = {
+    HudiQbeastSnapshot(tableID)
+  }
+
+  override def loadCurrentSchema(tableID: QTableID): StructType = {
+    loadSnapshot(tableID).schema
+  }
+
+  override def updateRevision(tableID: QTableID, revisionChange: RevisionChange): Unit = {}
+
+  override def updateTable(tableID: QTableID, tableChanges: TableChanges): Unit = {}
+
+  /**
+   * Returns the MetaClient for the table
+   * @param tableID
+   *   the table ID
+   * @return
+   */
+  private def loadMetaClient(tableID: QTableID): HoodieTableMetaClient = {
+    HoodieTableMetaClient
+      .builder()
+      .setConf(HadoopFSUtils.getStorageConfWithCopy(hadoopConf))
+      .setBasePath(tableID.id)
+      .build()
+  }
+
+  /**
+   * Checks if there's an existing log directory for the table
+   *
+   * @param tableID
+   *   the table ID
+   * @return
+   */
+  override def existsLog(tableID: QTableID): Boolean = {
+    val hoodiePath = new Path(tableID.id, HoodieTableMetaClient.METAFOLDER_NAME)
+    FileSystem.get(hadoopConf).exists(hoodiePath)
+  }
+
+  /**
+   * Creates an initial log directory
+   *
+   * @param tableID
+   *   Table ID
+   */
+  override def createLog(tableID: QTableID): Unit = {
+    if (!existsLog(tableID)) {
+      val hoodiePath = new Path(tableID.id, HoodieTableMetaClient.METAFOLDER_NAME)
+      FileSystem.get(hadoopConf).mkdirs(hoodiePath)
+    }
+  }
+
+  /**
+   * Creates the initial table files
+   *
+   * @param tableID
+   *   Table ID
+   */
+  private def createTable(tableID: QTableID, tableConfigs: Map[String, String]): Unit = {
+    val storageConfig = HadoopFSUtils.getStorageConfWithCopy(hadoopConf)
+    val properties = TypedProperties.fromMap(tableConfigs.asJava)
+
+    HoodieTableMetaClient
+      .withPropertyBuilder()
+      .setTableType(HoodieTableType.COPY_ON_WRITE)
```

Comment:
For now, I'm putting at the beginning the default properties that a table needs (if you write a new table in Hudi without any properties, it will create a new table with these properties). At the end, I'm overwriting these hardcoded properties with the user-provided properties/options using `.fromProperties(properties)`. This way, a user can provide more properties than the ones set here and can overwrite some of the ones set here.

In any case, it still needs to be fully evaluated which properties we should allow the user to provide and which not.